u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,136 @@
+
+            # Local maximum is indicated by signs_change == -2
+            if len(np.nonzero((signs_changes == -2))[0]) > 0:
+                m = np.nonzero((signs_changes == -2))[0][0] + 1
+            else:
+                return False, 0
+
+        # Check for local maximum when m is user defined
+        if np.nonzero((signs_changes == -2))[0][0] != m-1:
+            return False, m
+
+        # Remove r[0], the auto-correlation at order 0, that introduces bias.
+        r = r[1:]
+
+        stat = bartlett_formula(r, m - 1, len(ts))
+
+        # The upper limit of the significance interval.
+        band_upper = r.mean() + norm.ppf(1 - alpha/200)*r.var()
+
+        # Only local maximum with positive values are candidates to be tested against significance level.
+        season = r[m-1] > stat * band_upper
+
+        return season, m
+
+    return False, 0
+
+
+def bartlett_formula(r, m, length):
+    """
+    Computes the standard error of [r] at order [m] with respect to [length] according to Bartlett's formula.
+
+    :param r: The array whose standard error is to be computed.
+    :param m: The order of the standard error.
+    :param length: The size of the underlying sample to be used.
+    :return: The standard error of [r] with order [m].
+    """
+    if m == 1:
+        return math.sqrt(1/length)
+    else:
+        return math.sqrt((1 + 2 * sum(map(lambda x: x ** 2, r[:m-1]))) / length)
+
+
+def seasonal_adjustment(ts: 'TimeSeries', frequency, model='multiplicative') -> Tuple['TimeSeries', 'TimeSeries']:
+    """
+    Adjusts the TimeSeries [ts] for a seasonality of order [frequency] using the [model] decomposition.
+
+    :param ts: The TimeSeries to adjust.
+    :param frequency: The seasonality period.
+    :param model: -the type of decomposition to use (additive or multipilcative)
+    :return: A tuple (ts, seasonality) of TimeSeries where ts is the adjusted original TimeSeries ad seasonality
+            is a TimeSeries containing the extracted seasonality.
+    """
+    decomp = seasonal_decompose(ts.values(), model=model, freq=frequency)
+    seasonality = TimeSeries.from_times_and_values(ts.time_index(), decomp.seasonal)
+
+    if (seasonality.values() < 1e-6).any():
+        print("WARNING seasonal indexes equal to zero, using non-seasonal Theta method ")
+    else:
+        if model == 'multiplicative':
+            ts = ts / seasonality
+        else:
+            ts = ts - seasonality
+    return ts, seasonality
+
+
+def plot_acf(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha: float = 5):
+    """
+    Plots the ACF of [ts], highlighting the seasonality at lag [m] and the significance interval.
+
+    :param ts: The TimeSeries whose ACF should be plotted.
+    :param m:  The seasonality period.
+    :param max_lag: The maximal lag order to consider.
+    :param alpha: The confidence interval.
+    :return: Shows the plot.
```

Comment:
I'd remove this line, as the function doesn't return anything

Reply:
Indeed.